packages/widget_toolkit/doc/language-picker.md

```diff
@@ -43,56 +50,137 @@ your `pubspec.yaml` file.
 widget_toolkit: any
 ```
 
-The dependencies of the Language Picker should be provided in the root dependencies of the
-application:
-
-```dart
-MultiProvider(
-  providers: [
-    ...LanguagePickerDependencies.from(
-      LanguageServiceExample(localDataSource: _localDataSource),
-    ).providers, 
-  ],
-  child: this,
-);
-```
-
 After that you can import the package with the following line:
 
 `import 'package:widget_toolkit/language_picker.dart';`
 
-Additional step is the requirement to add the LanguagePickerTheme and LanguagePickerTheme as a 
-extension to your ThemeData.
+Additional step is the requirement to add the `LanguagePickerTheme` and `WidgetToolkitTheme` as  
+extensions to your `ThemeData`.
 
-as an example:
+Here is an example:
 ```dart
 //theme 
 //...
-extensions: [
-    darkMode ? WidgetToolkitTheme.dark : WidgetToolkitTheme.light,
-    darkMode ? LanguagePickerTheme.dark : LanguagePickerTheme.light,
-]
+MaterialApp(
+  theme: ThemeData(
+    extensions: [
+      darkMode ? WidgetToolkitTheme.dark : WidgetToolkitTheme.light,
+      darkMode ? LanguagePickerTheme.dark : LanguagePickerTheme.light,
+    ]
+  )
+);
 //..
 ```
 
+In order to fetch the models you want to present in the bottom sheet dialog you need to implement
+your `LanguageService` and/or extend `LanguageModel` and you can implement the `SelectedLanguageModel`:
+
+```dart
+class MyCustomLanguageModel implements LanguageModel {
+  MyCustomLanguageModel({
+    required this.locale,
+    required this.key,
+    required this.languageCode,
+  });
+
+  @override
+  final String locale;
+
+  @override
+  final String key;
+
+  @override
+  final String languageCode;
+
+  @override
+  String asString() => key;
+
+  @override
+  String withFirstCapitalLetter() => key.substring(0).toUpperCase();
+
+  @override
+  String translate(BuildContext context) => '';
```

Comment:
Please come up with a solution for language name translation

Reply:
The translation(model) function was provided to return the translated name of the language